ff2mpv.js

```diff
@@ -9,24 +9,30 @@ function ff2mpv(url) {
     browser.runtime.sendNativeMessage("ff2mpv", { url: url }).catch(onError);
 }
 
-browser.contextMenus.create({
-    id: "ff2mpv",
-    title: "Play in MPV"  + ( !!browser.contextMenus.getTargetElement ? " (&W)" : "" ),
-    contexts: ["link", "image", "video", "audio", "selection", "frame"]
-});
-
-browser.contextMenus.onClicked.addListener((info, tab) => {
-    switch (info.menuItemId) {
-        case "ff2mpv":
-            /* These should be mutually exclusive, but,
-               if they aren't, this is a reasonable priority.
-            */
-            url = info.linkUrl || info.srcUrl || info.selectionText || info.frameUrl;
-            if (url) ff2mpv(url);
-            break;
-    }
-});
-
-browser.browserAction.onClicked.addListener((tab) => {
-    ff2mpv(tab.url);
+// Make check on platform
+platform.then(function(i) {
```

Comment:
Let's use `browser.runtime.getPlatformInfo` here, since it's the MDN documented API.

Also, it's okay to `await` and assign the result somewhere. Calling `then` with a lambda here obscures the control flow for the core part of the extension.

Something roughly like this (not tested):

```js
let os = await browser.runtime.getPlatformInfo().then((i) => i.os);
```

Reply:
I forgot to paste the `var platform = browser.runtime.getPlatformInfo();`
I will update it. 

I test it yesterday on firefox windows, I will try to RHEL today.

Reply:
![Screenshot from 2021-11-04 09-08-17](https://user-images.githubusercontent.com/2817332/140245747-60740ecc-2697-4514-a96c-71308babf13c.png)
It's normal on RHEL 8.5

Reply:
Thanks for fixing that. Could you follow the rest of the advice above? I'd strongly prefer to not have the entire functional body of the plugin wrapped in a `then` application.

Reply:
Ok I'll look into it. Thanks for the advice

Reply:
> But the addons lifecycle is when firefox start up the function fire first so the addons can get the result right?

I'm not really sure -- this is the only extension I've ever written :-)

My justification for not using the `then` application is more that I find it harder to read, not that I'm worried about it being incorrect or buggy.

Reply:
> 
> 
> Let's use `browser.runtime.getPlatformInfo` here, since it's the MDN documented API.
> 
> Also, it's okay to `await` and assign the result somewhere. Calling `then` with a lambda here obscures the control flow for the core part of the extension.
> 
> Something roughly like this (not tested):
> 
> ```js
> let os = await browser.runtime.getPlatformInfo().then((i) => i.os);
> ```

This is not possible as then isn't a promise, how about 

```
var platform = browser.runtime.getPlatformInfo();

platform.then(function(i) {
  var os = i.os;
  var contextMenu =(os == "win") ? "&V" : "V (&W)";

  browser.contextMenus.create({
      id: "ff2mpv",
      title: "Play in MP"  + ( !!browser.contextMenus.getTargetElement ? contextMenu : "" ),
      contexts: ["link", "image", "video", "audio", "selection", "frame"]
  });
 
});
 
```

Reply:
> This is not possible as then isn't a promise, how about

Is it not? I'm not a very good JS programmer, but the [MDN says that `then()` returns a Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then).

(I'll play with the code in your branch when I get a free moment -- this might just be a misunderstanding on my part.)